### ISSUE TITLE
Handling RC versions

### ISSUE THREAD

**Issue by ximon18** [cat-in-136/cargo-generate-rpm#19]
Hi,

RPM [package versioning policy](https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/) supports the tilda character to denote a pre-release version. However, Cargo.toml doesn't permit a tilda in the version string.

With cargo-deb we have the same issue for DEB packaging policy but we pass the version in as a `--dev-version` command line argument when running `cargo deb`. With `cargo generate-rpm` I have to hack the Cargo.toml file _after_ I have run `cargo build` but _before_ I run `cargo generate-rpm` so that I can replace `-rcN` with `~rcN`.

Do you have any better suggestions on how to handle this scenario?

Thanks,

Ximon

**Comment by cat-in-136** [comment]
To specify different version than package.version, specify package.metadata.generate-rpm.version as following:

```toml
[package]
version = "0.1.0"

# ... snip ...

[package.metadata.generate-rpm]
version = "0.1.0~rc1"
```

Please refer README.md for other fields in package.metadata.generate-rpm

**Comment by ximon18** [comment]
Thanks.

However, unless I am missing something  that will unfortunately still require automated editing of the cargo.toml file as I do not want the package maintainer to have to manually remember to set the package version separately to the rust version.

Ximon

**Comment by cat-in-136** [comment]
@ximon18  I believe that it is resolved by #36.

**Comment by ximon18** [comment]
Yes, I'm using this new feature of v0.8.0 for exactly this, see https://github.com/NLnetLabs/routinator/pull/759.
